docs/dev/stream/state/queryable_state.md

```diff
@@ -32,38 +32,67 @@ under the License.
+eliminates the need for distributed operations/transactions with external systems such as key-value 
+stores which are often the bottleneck in practice. In addition, this feature may be particularly 
+useful for debugging purposes.
 
 <div class="alert alert-warning">
-  <strong>Attention:</strong> Queryable state accesses keyed state from a concurrent thread rather
-  than synchronizing with the operator and potentially blocking its operation. Since any state
-  backend using Java heap space, e.g. MemoryStateBackend or
-  FsStateBackend, does not work with copies when retrieving values but instead directly
-  references the stored values, read-modify-write patterns are unsafe and may cause the
-  queryable state server to fail due to concurrent modifications.
-  The RocksDBStateBackend is safe from these issues.
+  <strong>Attention:</strong> When querying a state object, that object is accessed from a concurrent 
+  thread without any synchronization or copying. This is a design choice, as any of the above would lead
+  to increased job latency, which we wanted to avoid. Since any state backend using Java heap space, 
+  <i>e.g.</i> <code>MemoryStateBackend</code> or <code>FsStateBackend</code>, does not work 
+  with copies when retrieving values but instead directly references the stored values, read-modify-write 
+  patterns are unsafe and may cause the queryable state server to fail due to concurrent modifications.
+  The <code>RocksDBStateBackend</code> is safe from these issues.
 </div>
 
+## Architecture
+
+Before showing how to use the Queryable State, it is useful to briefly describe the entities that compose it.
+The Queryable State consists of three main entities:
+
+ 1. the `QueryableStateClient`, which (potentially) runs outside the Flink cluster and submits the user queries, 
+ 2. the `QueryableStateClientProxy`, which runs on each `TaskManager` (*i.e.* inside the Flink cluster) and is responsible 
+ for receiving the client's queries, fetching the requested state on his behalf, and returning it to the client, and 
+ 3. the `QueryableStateServer` which runs on each `TaskManager` and is responsible for serving the locally stored state.
+ 
+In a nutshell, the client will connect to one of the proxies and send a request for the state associated with a specific 
+key, `k`. As stated in [Working with State]({{ site.baseurl }}/dev/stream/state/state.html), keyed state is organized in 
+*Key Groups*, and each `TaskManager` is assigned a number of these key groups. To discover which `TaskManager` is 
+responsible for the key group holding `k`, the proxy will ask the `JobManager`. Based on the answer, the proxy will 
+then query the `QueryableStateServer` running on that `TaskManager` for the state associated with `k`, and forward the
+response back to the client.
+
+## Activating Queryable State
+
+To enable queryable state on your Flink cluster, you just have to copy the 
```

Comment:
nit: omit "the"

Reply:
You mean make it: "To enable **the** queryable state on your Flink cluster..."?